From e5eab1d1ecb7c2323addba13989a2ec68d5c642d Mon Sep 17 00:00:00 2001
From: Juri Lelli <juri.lelli@gmail.com>
Date: Thu, 26 Apr 2012 09:57:20 +0200
Subject: [PATCH 20/48] use printk_sched were needed

Doing printk() and friends from scheduler context isn't actually
safe. We had a WARN_ON_ONCE to inform the user that the
replenishment were lagging to much. Substituted with a static
variable plus a printk_sched.
---
 kernel/sched/dl.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/kernel/sched/dl.c b/kernel/sched/dl.c
index dfd791a..5b3b257 100644
--- a/kernel/sched/dl.c
+++ b/kernel/sched/dl.c
@@ -318,7 +318,12 @@ static void replenish_dl_entity(struct sched_dl_entity *dl_se,
 	 * entity.
 	 */
 	if (dl_time_before(dl_se->deadline, rq->clock)) {
-		WARN_ON_ONCE(1);
+		static bool lag_once = false;
+
+		if (!lag_once) {
+			lag_once = true;
+			printk_sched("sched: DL replenish lagged to much\n");
+		}
 		dl_se->deadline = rq->clock + pi_se->dl_deadline;
 		dl_se->runtime = pi_se->dl_runtime;
 	}
-- 
1.7.9.5

